Add retries to FileTreeView.list

eatkins opened the pull request:
In 52cb29be7f911d73e501f52b76659d8efd6ad49c, we stopped catching all
exceptions in FileTreeView.list. This exposed an issue on windows where
sometimes an AccessDeniedException was thrown but what was quietly
ignored. To work around this, add retries when neither
NotDirectoryException nor NoSuchFileException are thrown. This is
really just to handle AccessDeniedExceptions on Windows.

eatkins committed:
Add retries to FileTreeView.list

In 52cb29be7f911d73e501f52b76659d8efd6ad49c, we stopped catching all
exceptions in FileTreeView.list. This exposed an issue on windows where
sometimes an AccessDeniedException was thrown but what was quietly
ignored. To work around this, add retries when neither
NotDirectoryException nor NoSuchFileException are thrown. This is
really just to handle AccessDeniedExceptions on Windows.

typesafe-tools commented:
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/parambuild/?sbt=develop&zinc=develop&io=pull/262/head&librarymanagement=develop&util=develop&website=develop&deploy_to_bintray=false&run_sbt_community=false&verbose_dbuild=false&community_verbose=false">this link.</A>

eatkins committed:
Drop all errors in IO.delete

There is an expectation that IO.delete never throws an IOException. On
windows, scripted tests sometimes try to delete directories that are
still in use. If we don't catch all IOExceptions here, scripted tests
will fail on Windows. We could catch the exception at the call site, but
it seems likely that there are builds/consumers of sbt that expect this
method to not throw.

eatkins commented:
On my vm, any scripted test that ran from the launcher would fail on windows with M14. Each of these commits independently pass.
